File: woodroid/src/main/java/me/gilo/woodroid/repo/CouponRepository.java
package me.gilo.woodroid.repo;

import me.gilo.woodroid.data.api.CouponAPI;
import me.gilo.woodroid.models.Coupon;
import me.gilo.woodroid.models.filters.CouponFilter;
import retrofit2.Call;

import java.util.List;

public class CouponRepository extends WooRepository{

    private final CouponAPI apiService;

    public CouponRepository(String baseUrl, String consumerKey, String consumerSecret) {
       super( baseUrl, consumerKey, consumerSecret);
       apiService = retrofit.create(CouponAPI.class);
    }

    public Call<Coupon> create(Coupon coupon) {
        return apiService.create(coupon);
    }


    public Call<Coupon> coupon(int id) {
        return apiService.view(id);
    }

    public Call<List<Coupon>> coupons() {
        return apiService.list();
    }

    public Call<List<Coupon>> coupons(CouponFilter couponFilter) {
        return apiService.filter(couponFilter.getFilters());
    }

    public Call<Coupon> update(int id, Coupon coupon) {
        return apiService.update(id, coupon);
    }

    public Call<Coupon> delete(int id) {
        return apiService.delete(id);
    }

    public Call<Coupon> delete(int id, boolean force) {
        return apiService.delete(id, force);
    }


}

File: woodroid/src/main/java/me/gilo/woodroid/repo/CouponRepository.kt
package me.gilo.woodroid.repo

import me.gilo.woodroid.data.api.CouponAPI
import me.gilo.woodroid.models.Coupon
import me.gilo.woodroid.models.filters.CouponFilter
import retrofit2.Call

class CouponRepository(baseUrl: String, consumerKey: String, consumerSecret: String) :
    WooRepository(baseUrl, consumerKey, consumerSecret) {

    private val apiService: CouponAPI

    init {
        apiService = retrofit.create(CouponAPI::class.java)
    }

    fun create(coupon: Coupon): Call<Coupon> {
        return apiService.create(coupon)
    }


    fun coupon(id: Int): Call<Coupon> {
        return apiService.view(id)
    }

    fun coupons(): Call<List<Coupon>> {
        return apiService.list()
    }

    fun coupons(couponFilter: CouponFilter): Call<List<Coupon>> {
        return apiService.filter(couponFilter.filters)
    }

    fun update(id: Int, coupon: Coupon): Call<Coupon> {
        return apiService.update(id, coupon)
    }

    fun delete(id: Int): Call<Coupon> {
        return apiService.delete(id)
    }

    fun delete(id: Int, force: Boolean): Call<Coupon> {
        return apiService.delete(id, force)
    }


}
